src/pages/RFOX/components/Stake/hooks/useRfoxStake.tsx

```diff
@@ -0,0 +1,334 @@
+    },
+  })
+
+  const stakeFeesQueryInput = useMemo(
+    () => ({
+      to: RFOX_PROXY_CONTRACT_ADDRESS,
+      accountNumber: stakingAssetAccountNumber,
+      data: stakeCallData,
+      value: '0',
+      chainId: fromAssetId(stakingAssetId).chainId,
+    }),
+    [stakeCallData, stakingAssetAccountNumber, stakingAssetId],
+  )
+
+  const isGetStakeFeesEnabled = useMemo(
+    () =>
+      Boolean(
+        stakeMutation.isIdle &&
+          hasEnoughBalance &&
+          isValidStakingAmount &&
+          runeAddress &&
+          !Boolean(errors?.amountFieldInput || errors?.manualRuneAddress) &&
+          allowanceQuery.isSuccess &&
+          !isApprovalRequired,
+      ),
+    [
+      allowanceQuery.isSuccess,
+      errors?.amountFieldInput,
+      errors?.manualRuneAddress,
+      hasEnoughBalance,
+      isApprovalRequired,
+      isValidStakingAmount,
+      runeAddress,
+      stakeMutation.isIdle,
+    ],
+  )
+
+  const stakeFeesQuery = useEvmFees({
+    ...stakeFeesQueryInput,
+    enabled: isGetStakeFeesEnabled,
+    staleTime: 30_000,
+    // Ensures fees are refetched at an interval, including when the app is in the background
+    refetchIntervalInBackground: true,
+    // Yeah this is arbitrary but come on, Arb is cheap
+    refetchInterval: 15_000,
+  })
+
+  const approvalMutation = useMutation({
+    ...reactQueries.mutations.approve({
+      assetId: stakingAssetId,
+      spender: RFOX_PROXY_CONTRACT_ADDRESS,
+      from: stakingAssetAccountAddress,
+      amountCryptoBaseUnit,
+      wallet,
+      accountNumber: stakingAssetAccountNumber,
+    }),
+    onSuccess: (txId: string) => {
+      setApprovalTxHash(txId)
+      toast({
+        title: translate('modals.send.transactionSent'),
+        description: (
+          <Text>
+            {stakingAssetFeeAsset?.explorerTxLink && (
+              <Link href={`${stakingAssetFeeAsset.explorerTxLink}${txId}`} isExternal>
+                {translate('modals.status.viewExplorer')} <ExternalLinkIcon mx='2px' />
+              </Link>
+            )}
+          </Text>
+        ),
+        status: 'success',
+        duration: 9000,
+        isClosable: true,
+        position: 'top-right',
+      })
+    },
+  })
+
+  const serializedApprovalTxIndex = useMemo(() => {
+    if (!(approvalTxHash && stakingAssetAccountAddress && stakingAssetAccountId)) return undefined
+
+    return serializeTxIndex(stakingAssetAccountId, approvalTxHash, stakingAssetAccountAddress)
+  }, [approvalTxHash, stakingAssetAccountAddress, stakingAssetAccountId])
+
+  const approvalTx = useAppSelector(state =>
+    serializedApprovalTxIndex ? selectTxById(state, serializedApprovalTxIndex) : undefined,
+  )
+
+  return {
+    allowanceQuery,
+    approvalFeesQuery,
+    isGetApprovalFeesEnabled,
+    allowanceCryptoPrecision,
+    isApprovalRequired,
+    stakeFeesQuery,
+    isGetStakeFeesEnabled,
+    approvalMutation,
+    approvalTxHash,
+    approvalTx,
+    stakeMutation,
+  }
```

Comment:
also might want to memoize this

Reply:
https://github.com/shapeshift/web/pull/7079#discussion_r1634369672